sql/core/src/main/scala/org/apache/spark/sql/ArrowConverters.scala

```diff
@@ -185,18 +225,19 @@ private[sql] abstract class PrimitiveColumnWriter(protected val allocator: BaseA
   protected var count = 0
   protected var nullCount = 0
 
-  override def init(initialSize: Int): Unit = {
+  override def init(initialSize: Option[Int]): this.type = {
+    initialSize.foreach(valueVector.setInitialCapacity)
```

Comment:
Ok, I'll remove the option to set an initial capacity now.  The default capacity seems to be fine.